include/envoy/ratelimit/ratelimit.h

```diff
@@ -39,21 +45,62 @@ struct Descriptor {
 };
 
 /**
+ * A single token bucket. See token_bucket.proto.
+ */
+struct TokenBucket {
+  uint32_t max_tokens_;
+  uint32_t tokens_per_fill_;
+  absl::Duration fill_interval_;
+};
+
+/**
+ * A token bucket operational state.
+ */
+struct TokenState {
+  mutable std::atomic<uint32_t> tokens_;
+  MonotonicTime fill_time_;
+};
+
+/**
+ * A single rate limit request descriptor. See ratelimit.proto.
+ * The descriptor entries constitute the primary key for local descriptors.
+ */
+struct LocalDescriptor {
+  std::vector<DescriptorEntry> entries_;
+  TokenBucket token_bucket_ = {};
+  TokenState* token_state_ = nullptr;
+
+  friend bool operator==(const LocalDescriptor& lhs, const LocalDescriptor& rhs) {
```

Comment:
Can you add some brief comments on why equality/hash only considers entries and doesn't consider the token attributes? It's not immediately clear.